### PR TITLE
Event Handler Service: Subscribe to all events (and don't subscribe in Consumer API anymore)

### PR DESCRIPTION
**Pull request by NikolaDmitrasinovic**
# Readiness checklist

-   [x] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description
Integration events are handled by the new service added at the top level of the application.

Changes made:
* Created new module at the top level: `EventHandlerService`
* Added both source and test projects
* Removed integration event handling from the `ConsumerApi`

### CODE REVIEW

File: IntegrationEventHandlerService/test/IntegrationEventHandlerService.Tests/IntegrationEventHandlerService.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
```
[the remark this answers is not in the document]
Comment (NikolaDmitrasinovic) [reply IntegrationEventHandlerService/test/IntegrationEventHandlerService.Tests/IntegrationEventHandlerService.Tests.csproj:6]: Done.

File: IntegrationEventHandlerService/test/IntegrationEventHandlerService.Tests/IntegrationEventHandlerServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+        // Act
+        await integrationEventHandlerService.StartAsync(CancellationToken.None);
+
+        // Assert
+        A.CallTo(() => mockEventBus.StartConsuming()).MustHaveHappenedOnceExactly();
+
+    }
+
+    private static IntegrationEventHandlerService CreateService(IEventBus? eventBus = null)
```
[the remark this answers is not in the document]
Comment (NikolaDmitrasinovic) [reply IntegrationEventHandlerService/test/IntegrationEventHandlerService.Tests/IntegrationEventHandlerServiceTests.cs:25]: There was some small refactoring going on here and for a second I thought that maybe I would need another test and yeah... that thing stayed there... 

Done.

File: IntegrationEventHandlerService/src/IntegrationEventHandlerService/IntegrationEventHandlerService.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    private void StartProcessing()
+    {
+        foreach (var module in _modules)
+        {
+            module.ConfigureEventBus(_eventBus);
+        }
+
+        _eventBus.StartConsuming();
+    }
```
[the remark this answers is not in the document]
Comment (NikolaDmitrasinovic) [reply IntegrationEventHandlerService/src/IntegrationEventHandlerService/IntegrationEventHandlerService.cs:37]: Much nicer, I like this.

Done.

File: IntegrationEventHandlerService/src/IntegrationEventHandlerService/Dockerfile
[the remark this answers is not in the document]
Comment (MH321Productions) [reply IntegrationEventHandlerService/src/IntegrationEventHandlerService/Dockerfile:1]: Done

**Review by tnotheis — DISMISSED** [review]

**Review by tnotheis — APPROVED** [review]

**Pull request merged**